### DISCUSSION TITLE
CenterAndExpand replacement

### DISCUSSION THREAD

**Discussion by danielftz** [dotnet/maui#5871] [Q&A]
What is the replacement for CenterAndExpand in MAUI? Currently I can't think of a single property that can be set to replicate the effect and it is thoroughly missed.

**Comment by hartez** [comment]
Well, first off, the option isn't gone. It's just deprecated. You can still use it in a StackLayout, if you really want.

But you can also replicate the AndExpand behavior with a Grid. 

```
<StackLayout>
	<Label Text="Top"/>
	<Label Text="Middle" VerticalOptions="CenterAndExpand"/>
	<Label Text="Bottom"/>
</StackLayout>
```

is equivalent to 

```
<Grid RowDefinitions="Auto,*,Auto">
	<Label Text="Top"/>
	<Label Text="Middle" Grid.Row="1" VerticalOptions="Center"/>
	<Label Text="Bottom" Grid.Row="2" />
</Grid>
```

**Reply by danielftz** [reply]
These two would not equivalent though. If "Middle" has its BackgroundColor set, having "CenterAndExpand" means the background color would expand to the entire Grid cell, and Center would not fill all the way to the border of the Grid cell. Same issue exists with any GestureRecognizer added to "Middle". My current workaround is adding a contentView around "Middle" or adding a BoxView underneath "Middle", neither are optimal compared to simply setting a property.

**Reply by hartez** [reply]
Can you clarify why they aren't equivalent? In a Forms application, this XAML:
```
<StackLayout>
    <Label Text="Top"/>
    <Label Text="Middle" VerticalOptions="CenterAndExpand" BackgroundColor="LightBlue"/>
    <Label Text="Bottom"/>
</StackLayout>
```

yields:

![stacklayout](https://user-images.githubusercontent.com/538025/162796785-730e3a00-acfe-4cf9-afad-9e98c9e79cb9.png)


And this XAML:

```
<Grid RowDefinitions="Auto,*,Auto">
    <Label Text="Top"/>
    <Label Text="Middle" Grid.Row="1" VerticalOptions="Center" BackgroundColor="LightGreen"/>
    <Label Text="Bottom" Grid.Row="2" />
</Grid>
```

yields

![grid](https://user-images.githubusercontent.com/538025/162796886-e785e62e-4e4a-42cd-bacb-ba621018ffeb.png)

The "Middle" Label occupies the same space in both scenarios. So I think I must be misunderstanding what you mean.
